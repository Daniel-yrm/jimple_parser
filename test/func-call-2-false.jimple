public final class FooKt extends java.lang.Object
{

    public static final int add_fourty_two(int)
    {
        int i0, $i1;

        i0 := @parameter0: int;

        $i1 = 42 + i0;

        return $i1;
    }

    public static final void main()
    {
        int $i0;
        java.lang.AssertionError $r0;
        boolean $z0, $z1;

        $i0 = staticinvoke <FooKt: int add_fourty_two(int)>(1);

        if $i0 != 44 goto label1;

        $z1 = 1;

        goto label2;

     label1:
        $z1 = 0;

     label2:
        $z0 = <kotlin._Assertions: boolean ENABLED>;

        if $z0 == 0 goto label3;

        if $z1 != 0 goto label3;

        $r0 = new java.lang.AssertionError;

        specialinvoke $r0.<java.lang.AssertionError: void <init>(java.lang.Object)>("Assertion failed");

        throw $r0;

     label3:
        return;
    }

    public static void main(java.lang.String[])
    {
        java.lang.String[] r0;

        r0 := @parameter0: java.lang.String[];

        staticinvoke <FooKt: void main()>();

        return;
    }
}
