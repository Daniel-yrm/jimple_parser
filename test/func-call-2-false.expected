{"extends":"java.lang.Object","object":"Class","name":"FooKt","modifiers":["public","final"],"content":[{"object":"Method","name":"add_fourty_two","modifiers":["public","static","final"],"content":[{"object":"Variable","name":"i0","type":{"identifier":"int","dimensions":0}},{"object":"Variable","name":"$i1","type":{"identifier":"int","dimensions":0}},{"object":"SetVariable","name":"i0","value":{"to":{"identifier":"int","dimensions":0},"expr_type":"cast","from":{"expr_type":"symbol","value":"@parameter0"}}},{"object":"SetVariable","name":"$i1","value":{"rhs":{"expr_type":"symbol","value":"i0"},"operator":"+","expr_type":"binop","lhs":{"expr_type":"constant","value":"42"}}},{"object":"Return","value":{"expr_type":"symbol","value":"$i1"}}],"type":{"identifier":"int","dimensions":0},"parameters":[{"identifier":"int","dimensions":0}]},{"object":"Method","name":"main","modifiers":["public","static","final"],"content":[{"object":"Variable","name":"$i0","type":{"identifier":"int","dimensions":0}},{"object":"Variable","name":"$r0","type":{"identifier":"java.lang.AssertionError","dimensions":0}},{"object":"Variable","name":"$z0","type":{"identifier":"boolean","dimensions":0}},{"object":"Variable","name":"$z1","type":{"identifier":"boolean","dimensions":0}},{"object":"SetVariable","name":"$i0","value":{"base_class":"FooKt","expr_type":"static_invoke","method":"add_fourty_two","parameters":[{"expr_type":"constant","value":"1"}]}},{"goto":"label1","object":"If","expression":{"rhs":{"expr_type":"constant","value":"44"},"operator":"notequal","expr_type":"binop","lhs":{"expr_type":"symbol","value":"$i0"}}},{"object":"SetVariable","name":"$z1","value":{"expr_type":"constant","value":"1"}},{"goto":"label2","object":"Goto"},{"label_id":"label1","object":"Label","content":[]},{"object":"SetVariable","name":"$z1","value":{"expr_type":"constant","value":"0"}},{"label_id":"label2","object":"Label","content":[]},{"object":"SetVariable","name":"$z0","value":{"expr_type":"field_access","from":"kotlin._Assertions","type":{"identifier":"boolean","dimensions":0},"field":"ENABLED"}},{"goto":"label3","object":"If","expression":{"rhs":{"expr_type":"constant","value":"0"},"operator":"==","expr_type":"binop","lhs":{"expr_type":"symbol","value":"$z0"}}},{"goto":"label3","object":"If","expression":{"rhs":{"expr_type":"constant","value":"0"},"operator":"notequal","expr_type":"binop","lhs":{"expr_type":"symbol","value":"$z1"}}},{"object":"SetVariable","name":"$r0","value":{"expr_type":"new","type":{"identifier":"java.lang.AssertionError","dimensions":0}}},{"base_class":"java.lang.AssertionError","parameter_type":[{"identifier":"java.lang.Object","dimensions":0}],"object":"SpecialInvoke","method":"<init>","variable":"$r0","parameters":[]},{"expr":{"expr_type":"symbol","value":"$r0"},"object":"Throw"},{"label_id":"label3","object":"Label","content":[]},{"object":"Return"}],"type":{"identifier":"void","dimensions":0},"parameters":[]},{"object":"Method","name":"main","modifiers":["public","static"],"content":[{"object":"Variable","name":"r0","type":{"identifier":"java.lang.String","dimensions":1}},{"object":"SetVariable","name":"r0","value":{"to":{"identifier":"java.lang.String","dimensions":1},"expr_type":"cast","from":{"expr_type":"symbol","value":"@parameter0"}}},{"base_class":"FooKt","object":"StaticInvoke","method":"main","parameters":[]},{"object":"Return"}],"type":{"identifier":"void","dimensions":0},"parameters":[{"identifier":"java.lang.String","dimensions":1}]}]}